client/src/Tools/_framework/ToolPanels/MoveContentToFolder.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useEffect, useState } from "react";
+import {
+  Text,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  SimpleGrid,
+  Heading,
+  IconButton,
+  HStack,
+} from "@chakra-ui/react";
+import axios from "axios";
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { useFetcher } from "react-router-dom";
+
+interface activeView {
+  folder: null | {
+    name: string;
+    id: number;
+    parentFolderId: number | null;
+  };
+  contents: {
+    isFolder: boolean;
+    name: string;
+    id: number;
+  }[];
+}
+
+export default function MoveContentToFolder({ isOpen, onClose, id }) {
+  // Set when the modal opens
+  const [parentId, setParentId] = useState<number | null>(null);
+  const [contentName, setContentName] = useState<string>("");
+  useEffect(() => {
+    if (isOpen) {
+      async function saveParentId() {
+        const {
+          data: { parentFolderId, name },
+        } = await axios.get(`/api/getMyContentInfo/${id}`);
+        updateActiveView(parentFolderId);
+        setParentId((_) => parentFolderId);
+        setContentName((_) => name);
+      }
+      saveParentId();
+    }
+  }, [isOpen]);
+
+  // Set whenever the user navigates to another folder
+  const [activeView, setActiveView] = useState<activeView>({
+    folder: null,
+    contents: [],
+  });
+  async function updateActiveView(newActiveFolderId: number | null) {
+    const {
+      data: {
+        folder: { content: contentFromApi },
+      },
+    } = await axios.get(`/api/getFolderContent/${newActiveFolderId ?? ""}`);
+
+    const content = contentFromApi
+      .map((item) => {
+        return {
+          isFolder: item.isFolder,
+          name: item.name,
+          id: item.id,
+        };
+      })
+      .sort((a, b) => {
+        if (a.isFolder && !b.isFolder) {
+          return -1;
+        } else if (b.isFolder && !a.isFolder) {
+          return 1;
+        } else {
+          return 0;
+        }
+      });
+
+    let folderData;
+    if (newActiveFolderId) {
+      const { data: contentInfo } = await axios.get(
```

Comment:
I'm confused why you are calling `getMyContentInfo` here. It seems that the above call to `getFolderContent` should have all the information you need.  If this is true, it doesn't look like you need `getMyContentInfo`. The api `/api/getFolderContent` gets only your own content, as it sets `ownerId` to `loggedInUserId`. The model function `getFolderContent` is more general, but it isn't used that way by the api.

Reply:
Ok, I deleted /api/getMyContentInfo. I just had to add the folder name to getFolderContent, which I think is fine, right?